packages/next-app/src/lib/gql.ts

```diff
@@ -1,16 +1,42 @@
 import { GraphQLClient } from "graphql-request";
-import {ENDPOINTS} from "@balancer-pool-metadata/balancer-gql/codegen";
-// import { getSdkWithHooks as mainnetSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
-import { getSdkWithHooks as goerliSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/goerli";
-// import { getSdkWithHooks as polygonSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/polygon";
-// import { getSdkWithHooks as arbitrumSdk  } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
+import {
+  ENDPOINTS,
+  Network,
+} from "@balancer-pool-metadata/balancer-gql/codegen";
+import { getSdkWithHooks as mainnetSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/mainnet";
+import { getSdkWithHooks as goerliSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/goerli";
+import { getSdkWithHooks as polygonSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/polygon";
+import { getSdkWithHooks as arbitrumSdk } from "@balancer-pool-metadata/balancer-gql/src/gql/__generated__/arbitrum";
 
+const currentNetwork =
+  typeof localStorage !== "undefined"
+    ? localStorage.getItem("networkId") ?? "1"
+    : "1";
 
-const client = new GraphQLClient(
-// TODO: replace by actual selected network
-ENDPOINTS.goerli
-);
+function networkFor(key: string | number): Network {
+  switch (key.toString()) {
+    case "1":
+      return Network.mainnet;
+    case "5":
+      return Network.goerli;
+    case "137":
+      return Network.polygon;
+    case "42161":
+      return Network.arbitrum;
+    default:
+      return Network.mainnet;
+  }
+}
 
-const gql = goerliSdk(client);
+const networkSdks = {
+  [Network.mainnet]: mainnetSdk,
+  [Network.polygon]: polygonSdk,
+  [Network.arbitrum]: arbitrumSdk,
+  [Network.goerli]: goerliSdk,
+};
 
-export default gql
\ No newline at end of file
+const client = new GraphQLClient(ENDPOINTS[networkFor(currentNetwork)]);
+
+const gql = networkSdks[networkFor(currentNetwork)](client);
```

Comment:
if this doesn't work what I'd probably do is wrap the gql client in a function that gets the selected network as a parameter and spits out the instantiated sdk

Reply:
I don't understand what you mean here, can you elaborate a little more?

Reply:
Meant to have something like `const getSdk = (network) => getSdkWithHooks(getClient(network))` but not an issue 👍